ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtil.java

```diff
@@ -547,9 +547,17 @@ private boolean isBellatrixBlockOld(final ReadOnlyStore store, final UInt64 bloc
         .isLessThanOrEqualTo(getCurrentSlot(store));
   }
 
-  private boolean isExecutionBlock(final ReadOnlyStore store, final Bytes32 blockRoot) {
+  private boolean isExecutionBlock(
+      final ReadOnlyStore store, final UInt64 blockSlot, final Bytes32 parentBlockRoot) {
```

Comment:
would be simpler to test block body version instead of slot?
`block.getBeaconBlock().map(BeaconBlock::getBody).map(BeaconBlockBody::toVersionCapella)`

Reply:
Hmm actually not sure, can it be blinded block?

Reply:
```java
    BeaconBlockBody body = block.getMessage().getBody();
    if(body.toVersionCapella().isPresent() || body.toBlindedVersionCapella().isPresent()) {
      return true;
    }
```
?

Reply:
looks good